### PR TITLE
Replace more usages of assertTrue/False

### PR DESCRIPTION
**Pull request by Flamefire**
I found yet another bug in the tests in `test_pypi_source_urls`:

> `self.assertTrue(eb340_url, res)`

This test will always succeed, `assertIn` was intended.

Hence I did another RegEx-powered Search&Replace and introduced `assert(Not)Exists` to replace the most common use of `assertTrue/False` which also fixes a few bugs in error messages.

### CODE REVIEW

File: test/framework/filetools.py [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ def test_back_up_file(self):
         new_file = [x for x in test_files if x not in known_files][0]
         self.assertTrue(new_file.startswith('test.txt.bak_'))
         known_files = os.listdir(os.path.dirname(fp))
-        self.assertTrue(ft.read_file(first_normal_backup), txt)
+        self.assertEqual(ft.read_file(first_normal_backup), txt)
```
Comment (jfgrimm) [inline test/framework/filetools.py:1012]: these change the behaviour, but that looks intended

File: test/framework/filetools.py [hunk 2]
```diff
@@ -2313,24 +2313,25 @@ def test_remove(self):
...
-            self.assertFalse(os.path.exists(test_dir) or os.path.isdir(test_dir))
+            self.assertNotExists(test_dir)
 
         # remove also takes a list of paths
         ft.write_file(testfile, 'bar')
         ft.mkdir(test_dir)
-        self.assertTrue(os.path.exists(testfile))
-        self.assertTrue(os.path.exists(test_dir) and os.path.isdir(test_dir))
+        self.assertExists(testfile)
+        self.assertExists(test_dir) and os.path.isdir(test_dir)
```
Comment (jfgrimm) [inline test/framework/filetools.py:2331]:
```suggestion
        self.assertExists(test_dir)
        self.assertTrue(os.path.isdir(test_dir))
```

File: test/framework/systemtools.py [hunk 3]
```diff
@@ -401,7 +401,7 @@ def tearDown(self):
     def test_avail_core_count_native(self):
         """Test getting core count."""
         core_count = get_avail_core_count()
-        self.assertTrue(isinstance(core_count, int), "core_count has type int: %s, %s" % (core_count, type(core_count)))
+        self.assertIsInstance(core_count, int), "core_count has type int: %s, %s" % (core_count, type(core_count))
```
Comment (jfgrimm) [inline test/framework/systemtools.py:404]: wrong brackets
```suggestion
        self.assertIsInstance(core_count, int, "core_count has type int: %s, %s" % (core_count, type(core_count)))
```

**Review by jfgrimm — CHANGES_REQUESTED** [review]

File: test/framework/easyblock.py [hunk 4]
```diff
@@ -1783,7 +1783,7 @@ def test_obtain_file(self):
             if res is not None:
                 loc = os.path.join(tmpdir, 't', 'toy', fn)
                 self.assertEqual(res, loc)
-                self.assertTrue(os.path.exists(loc), "%s file is found at %s" % (fn, loc))
+                self.assertExists(loc), "%s file is found at %s" % (fn, loc)
```
[the remark this answers is not in the document]
Comment (Flamefire) [reply test/framework/easyblock.py]: I actually intended to remove the message. It doesn't provide any value over the default (`fn` is part of `loc`)

File: test/framework/filetools.py (hunk 1 above)
Comment (Flamefire) [reply test/framework/filetools.py:1012]: Yes, this was a bug before.

File: test/framework/systemtools.py (hunk 3 above)
Comment (Flamefire) [reply test/framework/systemtools.py]: I missed to remove the message. Thanks!

**Review by jfgrimm — APPROVED** [review]
lgtm

**Comment by jfgrimm** [comment]
Going in, thanks @Flamefire!

**Pull request merged**